account_usage.ml

```diff
@@ -0,0 +1,362 @@
+let account_usage_layout policy unikernels blocks =
+  let deployed_unikernels = List.length unikernels in
+  let total_volume_used =
+    List.fold_left (fun total_size (_, size, _) -> total_size + size) 0 blocks
+  in
+  let total_free_space =
+    Option.value ~default:0 policy.Vmm_core.Policy.block - total_volume_used
+  in
+  let total_memory_used =
+    List.fold_left
+      (fun total_memory (_, unikernel) ->
+        total_memory + unikernel.Vmm_core.Unikernel.memory)
+      0 unikernels
+  in
+  let count_cpuid_usage =
+    let cpuid_count = Hashtbl.create (List.length unikernels) in
+    (* count usage of each cpuid in unikernels *)
+    List.iter
+      (fun (_, unikernel) ->
+        let cpuid = unikernel.Vmm_core.Unikernel.cpuid in
+        let count =
+          Hashtbl.find_opt cpuid_count cpuid |> Option.value ~default:0
+        in
+        Hashtbl.replace cpuid_count cpuid (count + 1))
+      unikernels;
+    (* Prepare list of all cpuids from policy *)
+    let policy_cpuids = Vmm_core.IS.to_list policy.cpuids in
+    (* Generate the list with all policy.cpuids and their respective counts *)
+    List.map
+      (fun cpuid ->
+        let count =
+          Hashtbl.find_opt cpuid_count cpuid |> Option.value ~default:0
+        in
+        (cpuid, count))
+      policy_cpuids
+  in
+  let count_bridge_usage =
+    let bridge_count =
+      Hashtbl.create (Vmm_core.String_set.cardinal policy.bridges)
+    in
+    (* Initialize all policy bridges with a count of 0 *)
+    Vmm_core.String_set.iter
+      (fun bridge -> Hashtbl.replace bridge_count bridge 0)
+      policy.bridges;
+
+    (* Count each bridge usage from unikernel.bridges, but only if it's in policy.bridges *)
+    List.iter
+      (fun (_, unikernel) ->
+        List.iter
+          (fun (name, bridge, _) ->
+            let bridge_name = Option.value bridge ~default:name in
+            (* Only count if the bridge is in policy.bridges *)
+            if Vmm_core.String_set.mem bridge_name policy.bridges then
+              let count =
+                Hashtbl.find_opt bridge_count bridge_name
+                |> Option.value ~default:0
+              in
+              Hashtbl.replace bridge_count bridge_name (count + 1))
+          unikernel.Vmm_core.Unikernel.bridges)
+      unikernels;
+
+    (* Convert the hash table to a sorted list of (bridge_name, count) *)
+    let bridge_usage_list =
+      Hashtbl.fold
+        (fun bridge count acc -> (bridge, count) :: acc)
+        bridge_count []
+    in
+    (* Sort the list by bridge name *)
+    List.sort (fun (b1, _) (b2, _) -> String.compare b1 b2) bridge_usage_list
+  in
```

Comment:
with the above code, I struggle a bit: there's non-trivial counting and lots of data structures involved. Maybe best to put such code in separate functions. I still don't grasp what the data structures for the UI should be? sorted lists?

Especially, I'm not a big fan of hashtable.

All of that shouldn't block this PR from being merged, but we may want to refactor later.

Reply:
My idea was to have a chart of the different bridges assigned to the user, and how many times each bridge is used. 
It's the same with cpuid.

If there's a more trivial way to get this information I'll be so happy to refactor